mathics/eval/drawing/plot.py

```diff
@@ -134,7 +134,7 @@ def quiet_f(*args):
 
 
 def eval_ListPlot(
-    plot_groups: list,
```

Comment:
Should it not be better to ensure using one type or the other?

Reply:
Hmm...  

plot_groups is sometimes the result of `expr.to_python()`. For a `ListExpression` this is a list.
In the case of constant data we have a tuple; here, this is synonymous with an immutable list. 

I think plot_groups should be a tuple - immutable. However, I don't think we can change the return type of `to_python()` and I don't think we can change it to tuple on return without first detecting that it is a list. Otherwise, we would be wrapping a scalar into a tuple. 

So, while I agree it would be better to ensure a tuple is passed in, at this stage, I am not sure I want take the risk of converting a list to tuple right now. 

I have made note of this issue though in commit 016f6be125b97f2e0a2329e28999495f2464bda6

Reply:
OK, I think it would be enough by now.